src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1927,6 +1995,70 @@ export class ConfigureWorkspaceFolderRecommendedExtensionsAction extends Abstrac
 	}
 }
 
+export class AddToWorkspaceRecommendationsAction extends AbstractConfigureRecommendedExtensionsAction {
+
+	static readonly ID = 'workbench.extensions.action.addToWorkspaceRecommendations';
+	static LABEL = localize('addToWorkspaceRecommendations', "Add to workspace recommendations");
+
+	constructor(
+		id: string,
+		label: string,
+		@IFileService fileService: IFileService,
+		@IWorkspaceContextService contextService: IWorkspaceContextService,
+		@IEditorService editorService: IEditorService,
+		@IJSONEditingService jsonEditingService: IJSONEditingService,
+		@ITextModelService textModelResolverService: ITextModelService,
+		@ICommandService private commandService: ICommandService,
+		@INotificationService private notificationService: INotificationService
+	) {
+		super(
+			id,
+			label,
+			contextService,
+			fileService,
+			editorService,
+			jsonEditingService,
+			textModelResolverService
+		);
+	}
+
+	run(): TPromise<void> {
+		if (!(this.editorService.activeEditor instanceof ExtensionsInput) || !this.editorService.activeEditor.extension) {
+			return TPromise.as(null);
+		}
+		const folders = this.contextService.getWorkspace().folders;
+		if (!folders || !folders.length) {
+			this.notificationService.info(localize('AddToWorkspaceRecommendations.noWorkspace', 'There is no workspace open to add recommendations.'));
+			return TPromise.as(null);
+		}
+
+		const extensionId = this.editorService.activeEditor.extension.id;
+		const pickFolderPromise = folders.length === 1
+			? TPromise.as(folders[0])
+			: this.commandService.executeCommand<IWorkspaceFolder>(PICK_WORKSPACE_FOLDER_COMMAND_ID);
+		return pickFolderPromise
+			.then(workspaceFolder => {
+				if (!workspaceFolder) {
+					return TPromise.as(null);
+				}
+				const configurationFile = workspaceFolder.toResource(paths.join('.vscode', 'extensions.json'));
+				return this.getFolderRecommendedExtensions(configurationFile).then(recommendations => {
+					if (recommendations.map(e => e.toLowerCase()).indexOf(extensionId.toLowerCase()) > -1) {
```

Comment:
This will iterate twice and create a temporary array in memory, that's why I went with `Array.prototype.some()`. Again, pretty insignificant footprint though

Reply:
fixed